mern/server/utils/parse.js

```diff
@@ -1,42 +1,72 @@
-import QA from "../models/question_answer.js"
+import QA from "../models/question_answer.js";
 
-export default function parse(transcript){
-
-    const lst = []
-    for (var i = 0; i<transcript.length; i++){
-
-        lst.push(createQAs(transcript[i]))
-    }
-
-    return lst
+//Parses the transcript by looking through each 'list' of questions and creating question answer pairs
+//See trancript JSON formatting for more details
+export default function parse(transcript) {
+  const lst = [];
+  for (var i = 0; i < transcript.length; i++) {
+    createQAs(transcript[i]);
+    lst.push(initialQAID(transcript[i]));
+  }
+  return lst;
 }
 
-function createQAs(qlist){
-
-    const initial = qlist[0].id
-
-    //qlist is a list of JSON objects that exists as questions/answer pairs
-    for (var i = 0; i <qlist.length; i++){
-
-        createQA(qlist[i])
-    }
-
-    // Return the first item in the list which should be the top level question
-    return initial
+//Loops through all questions in the JSON object qlist, making them into objects in the databse
+export function createQAs(qlist) {
+  //qlist is a list of JSON objects that exists as questions/answer pairs
+  for (var i = 0; i < qlist.length; i++) {
+    createQA(qlist[i]);
+  }
 }
 
-function createQA(item){
+// Returns a list of the initial QA ids
+export function initialQAID(qlist) {
+  if (qlist.length == 0) {
+    return null;
+  }
+  return qlist[0].id; // Return the first item in the list which should be the top level question
+}
 
-    const id = item.id
-    const question = item.question
-    const question_included = item.question_included
-    const intents = item.intents
-    
-    QA.create({
+//Creates a QA item in the database
+export function createQA(item) {
+  try {
+    const id = item.id;
+    const question = item.question;
+    const question_included = item.question_included;
+    const intents = item.intents;
+    // Checks to see that all subvalues are defined and question id is unique
+    if (isValidItem(item)) {
+      QA.create({
         id,
         question,
         question_included,
-        intents
-    })
+        intents,
+      });
+      return true;
+    }
+    return false;
+  } catch (e) {
+    console.log(e.message);
+    return false;
+  }
 }
 
+export function isValidItem(item) {
+  const id = item.id;
+  const question = item.question;
+  const question_included = item.question_included;
+  const intents = item.intents;
+  if (
+    typeof id === "undefined" ||
+    typeof question === "undefined" ||
+    typeof question_included === "undefined" ||
+    typeof intents === "undefined" ||
+    typeof intents.value === "undefined" ||
+    typeof intents.included === "undefined" ||
+    typeof intents.children === "undefined" ||
```

Comment:
If an item does not have children (but has all the other information) should it still be considered invalid? Or are we doing this because we are making our own dummy data thus we can assume all items includes all elements?